### PR TITLE
Bug 1398889 - Add param 'silent_users' that never trigger sending bug…

### PR DESCRIPTION
**Pull request by dylanwh**
…mail

### CODE REVIEW

**Comment by emceeaich** [comment]
UI looked fine on Mozilla-dev but couldn't test bugmail blocking there

File: Bugzilla/User.pm [hunk 1]
```diff
@@ -2407,13 +2409,24 @@ sub is_insider {
 sub is_global_watcher {
     my $self = shift;
 
-    if (!defined $self->{'is_global_watcher'}) {
-        my @watchers = split(/[,;]+/, Bugzilla->params->{'globalwatchers'});
-        $self->{'is_global_watcher'} = scalar(grep { $_ eq $self->login } @watchers) ? 1 : 0;
+    if (!exists $self->{'is_global_watcher'}) {
+        my @watchers = split(/\s*,\s*/, Bugzilla->params->{'globalwatchers'});
```
Comment (dylanwh) [inline Bugzilla/User.pm]: note: it was never documented that a ; could be used, so I removed it. Also added optional whitespace

File: template/en/default/admin/params/mta.html.tmpl [hunk 2]
```diff
@@ -73,6 +73,8 @@
                "Set to 0 to disable whining.",
 
   globalwatchers => "A comma-separated list of users who should receive a " _
-                    "copy of every notification mail the system sends." }
+                    "copy of every notification mail the system sends.",
+
+  silent_users => "A comma-separated list of users who never trigger sending email" }
```
Comment (globau) [inline template/en/default/admin/params/mta.html.tmpl]: nit: missing period at end of sentence.

File: Bugzilla/User.pm [hunk 3]
```diff
@@ -2407,13 +2409,24 @@ sub is_insider {
...
     }
     return  $self->{'is_global_watcher'};
 }
 
+sub is_silent_user {
+    my $self = shift;
+
+    if (!exists $self->{'is_silent_user'}) {
+        my @helpers = split(/\s*,\s*/, Bugzilla->params->{'silent_users'});
+        $self->{'is_silent_user'} = (any { $self->login eq $_ } @helpers) ? 1 : 0;
```
Comment (globau) [inline Bugzilla/User.pm]: `helpers` is a weird name to use here, instead of something like `users`.

File: Bugzilla/User.pm [hunk 4]
```diff
@@ -2246,6 +2246,8 @@ sub wants_bug_mail {
     my $self = shift;
     my ($bug, $relationship, $fieldDiffs, $comments, $dep_mail, $changer) = @_;
 
+    return 0 if $changer && $changer->is_silent_user;
```
Comment (globau) [inline Bugzilla/User.pm]: i feel this needs a comment which points to the param.

File: Bugzilla/User.pm (hunk 3 above)
Comment (dylanwh) [reply Bugzilla/User.pm]: woops. Originally it was "global_helpers" before I settled on silent_users

**Comment by dylanwh** [comment]
Fixed. Also guess what! globalwatchers was split on a DIFFERENT regex! I updated BugMail.pm to be consistent.

**Pull request merged**